coresdk/src/test/unit_tests/unit_test_bitmap.cpp

```diff
@@ -128,3 +128,62 @@ TEST_CASE("bitmap bounding details can be retrieved", "[bitmap]")
     }
     free_bitmap(bmp);
 }
+
+TEST_CASE("name can be retrieved from bitmap", "[bitmap][bitmap_name]")
+{   
+    SECTION("can get bitmap name")
+    {
+        bitmap bmp = load_bitmap("rocket", "rocket_sprt.png");
+        REQUIRE(bitmap_valid(bmp));
+        
+        string name = bitmap_name(bmp);
+        REQUIRE(name == "rocket");
+    }
+
+    SECTION("empty string returned for bitmap name when passing nullptr")
+    {
+        string name = bitmap_name(nullptr);
+        REQUIRE(name == "");
+    }
+
+    SECTION("empty string returned for bitmap when passing freed bitmap")
+    {
+        bitmap bmp = load_bitmap("frog", "frog.png");
+        REQUIRE(bitmap_valid(bmp));
+        
+        free_bitmap(bmp);
+
+        string name = bitmap_name(bmp);
+        REQUIRE_FALSE(name == "frog");
+        REQUIRE(name == "");
+    }
+
+    SECTION("can get unique names from bitmaps when names are reused")
+    {
+        bitmap bmp_1 = create_bitmap("blank", 300, 200);
+        bitmap bmp_2 = create_bitmap("blank", 800, 600);
+
+        string name_1 = bitmap_name(bmp_1);
+        string name_2 = bitmap_name(bmp_2);
+
+        REQUIRE(name_1 == "blank");
+        REQUIRE(name_2 == "blank0");
+    }
+
+    SECTION("can get name of bitmap when name is empty string")
+    {
+        bitmap bmp = create_bitmap("", 64, 64);
+        string name = bitmap_name(bmp);
+
+        REQUIRE(name == "");
+    }
+
+    SECTION("can get name of bitmap when name includes special characters")
+    {
+        string silly_name = "AbCdEf 2+3=5 !@#$% \n";
+        bitmap bmp = create_bitmap(silly_name, 128, 128);
+
+        string returned_name = bitmap_name(bmp);
+        REQUIRE(returned_name == silly_name);
```

Comment:
You can also use [Catch2's string matchers](https://github.com/catchorg/Catch2/blob/v2.x/docs/matchers.md) for assertions with std::strings.

For example, to use Equals:
```cpp
using Catch::Matchers::Equals;
```

```suggestion
        REQUIRE_THAT(returned_name, Equals(silly_name));
```
I'm not sure if there's a reason to prefer using one over the other, just thought it was worth noting. So don't consider this a required change - I'm half reminding myself here for things to note down in the guide!